### PR TITLE
Bugfixes in initializers

### PR DESCRIPTION
**Pull request by joglekara**
Also 
- adds smoothing to Arbitrary1V calculations
- Mora-Yahi distribution initialization
- Bugfix for arts1v test

### CODE REVIEW

**Comment by joglekara** [comment]
Also implements 3 different Spherical Harmonic fitting routines

f0 - always DLM
f10, f11 can now be
- arbitrary + smoothing
- Mora-Yahi (so just gradient scale length as the free parameter)
- neural network

File: tests/configs/arts2d_test_inputs.yaml [hunk 1]
```diff
@@ -75,28 +75,30 @@ parameters:
       angle: 0.0
       lb: -10.0
       ub: 30.0
-      val: 5.0
+      val: 0.0
   electron:
     Te:
+      active: true
```
Comment (almilder) [inline tests/configs/arts2d_test_inputs.yaml:81]: Duplicate line?

Comment (almilder) [reply tests/configs/arts2d_test_inputs.yaml:81]: Multiple in this file

File: tests/test_form_factor/test_epw.py [hunk 2]
```diff
@@ -1,16 +1,20 @@
 from jax import config, jit, numpy as jnp
 from copy import deepcopy
 import yaml, mlflow, tempfile, numpy as np
+from jax import config, jit, numpy as jnp
+from copy import deepcopy
+import yaml, mlflow, tempfile, numpy as np
```
Comment (almilder) [inline tests/test_form_factor/test_epw.py:6]: Duplicates

File: tests/test_forward/test_angular_2d.py [hunk 3]
```diff
@@ -91,8 +91,48 @@ def test_arts2d_forward_pass():
 
         mlflow.log_metric("runtime-sec", time.time() - t0)
 
-        # np.testing.assert_allclose(ThryE, ground_truth, rtol=1e-4)
-    misc.export_run(run.info.run_id)
+        np.testing.assert_allclose(ThryE, ground_truth, rtol=1e-4)
+
+
+def plot_fwd_vs_ground_truth(td, ts_params, ThryE, ground_truth):
```
Comment (almilder) [inline tests/test_forward/test_angular_2d.py:97]: Is this not redundant with the plotters from plotters?

File: tests/test_inverse/test_arts1d_random.py [hunk 4]
```diff
@@ -171,30 +202,25 @@ def loss_fn(_diff_params, _static_params):
...
-            # mlflow.log_metric(f"first run time", time.time() - t0)
-
             # dump ground truth to disk
-
             loss = 1
             while np.nan_to_num(loss, nan=1) > 5e-2:
                 # ts_diag = ThomsonScatteringDiagnostic(config, scattering_angles=sas)
                 diff_params, static_params = perturb_and_split_params(arbitrary_distribution, config, rng)
                 use_optax = True
+                use_optax = False
```
Comment (almilder) [inline tests/test_inverse/test_arts1d_random.py:212]: duplicate?

File: tests/test_inverse/test_arts1d_random.py [hunk 5]
```diff
@@ -171,30 +202,25 @@ def loss_fn(_diff_params, _static_params):
...
                 if use_optax:
 
+                    opt = optax.adam(1e-2)  # if arbitrary_distribution else 1e-2)
```
Comment (almilder) [inline tests/test_inverse/test_arts1d_random.py:215]: duplicate

File: tests/test_inverse/test_arts1d_random.py [hunk 6]
```diff
@@ -171,30 +202,25 @@ def loss_fn(_diff_params, _static_params):
...
+
                     opt = optax.adam(1e-2)  # if arbitrary_distribution else 1e-2)
                     opt_state = opt.init(diff_params)
                     for i in (pbar := tqdm.tqdm(range(1000))):
                         t0 = time.time()
                         loss, grad_loss = jit_vg(diff_params, static_params)
                         mlflow.log_metrics({f"iteration time": time.time() - t0, "loss": float(loss)}, step=i)
+                        mlflow.log_metrics({f"iteration time": time.time() - t0, "loss": float(loss)}, step=i)
```
Comment (almilder) [inline tests/test_inverse/test_arts1d_random.py:223]: duplicate

Comment (almilder) [reply tests/test_inverse/test_arts1d_random.py:223]: check whole file there are more

File: tsadar/inverse/loops.py [hunk 7]
```diff
@@ -18,11 +18,13 @@
...
 
 def _1d_scipy_loop_(
-    config: Dict, loss_fn: LossFunction, previous_weights, batch: Dict
+    config: Dict, loss_fn: LossFunction, previous_weights: np.ndarray, batch: Dict
 ) -> Tuple[float, Dict]:
+    _activate = True
     _activate = True
     if previous_weights is None:  # if prev, then use that, if not then use flattened weights
-        ts_params = ThomsonParams(config["parameters"], config["optimizer"]["batch_size"], activate= _activate)
+        ts_params = ThomsonParams(config["parameters"], config["optimizer"]["batch_size"], activate=_activate)
```
Comment (almilder) [inline tsadar/inverse/loops.py:26]: duplicate

File: tsadar/inverse/loops.py [hunk 8]
```diff
@@ -46,13 +48,13 @@ def _1d_scipy_loop_(
...
 def _1d_adam_loop_(
-    config: Dict, loss_fn: LossFunction, previous_weights, batch: Dict, tbatch
+    config: Dict, loss_fn: LossFunction, previous_weights: np.ndarray, batch: Dict, tbatch
 ) -> Tuple[float, Dict]:
 
     opt = optax.adam(config["optimizer"]["learning_rate"])
-    #ts_params = ThomsonParams(config["parameters"], config["optimizer"]["batch_size"])
     if previous_weights is None:  # if prev, then use that, if not then use flattened weights
         ts_params = ThomsonParams(config["parameters"], config["optimizer"]["batch_size"], activate=True)
+        ts_params = ThomsonParams(config["parameters"], config["optimizer"]["batch_size"], activate=True)
```
Comment (almilder) [inline tsadar/inverse/loops.py:57]: duplicate

**Review by almilder — CHANGES_REQUESTED** [review]
Lets remove the duplicate code and then as long as it passes the tests its good to go

**Comment by joglekara** [comment]
Done, looks like we're passing FormFactor and inverse tests. Need to fix forward pass tests as discussed but that will be a separate PR

**Review by almilder — APPROVED** [review]
